examples/with-javascript/src/commands/General/eval.js

```diff
@@ -0,0 +1,77 @@
+const { Command } = require('@sapphire/framework');
+const { send } = require('@sapphire/plugin-editable-commands');
+const { Type } = require('@sapphire/type');
+const { codeBlock, isThenable } = require('@sapphire/utilities');
+const { inspect } = require('util');
+
+class UserCommand extends Command {
+	constructor(context, options) {
+		super(context, {
+			...options,
+			aliases: ['ev'],
+			description: 'Evals any JavaScipt code',
+			quotes: [],
+			preconditions: ['OwnerOnly'],
+			flags: ['async', 'hidden', 'showHidden', 'silent', 's'],
+			options: ['depth']
+		});
+	}
+
+	async run(message, args) {
+		const code = await args.rest('string');
+
+		const { result, success, type } = await this.eval(message, code, {
+			async: args.getFlags('async'),
+			depth: Number(args.getOption('depth')) ?? 0,
+			showHidden: args.getFlags('hidden', 'showHidden')
+		});
+
+		const output = success ? codeBlock('js', result) : `**ERROR**: ${codeBlock('bash', result)}`;
+		if (args.getFlags('silent', 's')) return null;
+
+		const typeFooter = `**Type**: ${codeBlock('typescript', type)}`;
+
+		if (output.length > 2000) {
+			return send(message, {
+				content: `Output was too long... sent the result as a file.\n\n${typeFooter}`,
+				files: [{ attachment: Buffer.from(output), name: 'output.js' }]
+			});
+		}
+
+		return send(message, `${output}\n${typeFooter}`);
+	}
+
+	async eval(message, code, flags) {
+		if (flags.async) code = `(async () => {\n${code}\n})();`;
+
+		const msg = message;
```

Comment:
This doesn't seem to be used anywhere in the code, was this a mistake or?

Reply:
I'm not entirely sure. For the most part I just dumped the code, stripped the types, and converted things where appropriate. Any mistakes in here are from the TS version - but keen eye, I'll go over the code today.

Reply:
It's so you can do `!ev msg.whatever` as well as `!ev message.channel`. `eval` has access to all parameters within the current scope, and the parameter here is called `message`.